FreeSurfer/LongitudinalFreeSurferPipeline.sh

```diff
@@ -0,0 +1,350 @@
+  else
+    log_Err_Abort "Cannot tell which version of FreeSurfer you are using."
+  fi
+
+  # strip out extraneous stuff from FreeSurfer version string
+  freesurfer_version_string_array=(${freesurfer_version_string//-/ })
+  freesurfer_version=${freesurfer_version_string_array[5]}
+  freesurfer_version=${freesurfer_version#v} # strip leading "v"
+
+  log_Msg "INFO: Determined that FreeSurfer version is: ${freesurfer_version}"
+
+  # break FreeSurfer version into components
+  # primary, secondary, and tertiary
+  # version X.Y.Z ==> X primary, Y secondary, Z tertiary
+  freesurfer_version_array=(${freesurfer_version//./ })
+
+  freesurfer_primary_version="${freesurfer_version_array[0]}"
+  freesurfer_primary_version=${freesurfer_primary_version//[!0-9]/}
+
+  freesurfer_secondary_version="${freesurfer_version_array[1]}"
+  freesurfer_secondary_version=${freesurfer_secondary_version//[!0-9]/}
+
+  freesurfer_tertiary_version="${freesurfer_version_array[2]}"
+  freesurfer_tertiary_version=${freesurfer_tertiary_version//[!0-9]/}
+
+  if [[ $(( ${freesurfer_primary_version} )) -lt 6 ]]; then
+    # e.g. 4.y.z, 5.y.z
+    log_Err_Abort "FreeSurfer version 6.0.0 or greater is required. (Use FreeSurferPipeline-v5.3.0-HCP.sh if you want to continue using FreeSurfer 5.3)"
+  fi
+}
+
+# Configure the use of FreeSurfer v6 custom tools
+configure_custom_tools
+
+# Show tool versions
+show_tool_versions
+
+# Validate version of FreeSurfer in use
+validate_freesurfer_version
+
+#processing code goes here
+echo "parallel mode: $parallel_mode"
+fslsub_queue=NONE
+if [ "$parallel_mode" == "FSLSUB" ]; then
+  if [ -n "$queue" ]; then fslsub_queue=$queue; fi
+elif [ "$parallel_mode" != "NONE" -a "$parallel_mode" != "BUILTIN" ] then 
+  log_Err_Abort "Unknown parallel mode $parallel_mode. Plese specify one of FSLSUB, BUILTIN, NONE"
+fi
+
+start_stage=0
+if [ -n "$StartStage" ]; then  
+  case $StartStage in
+    TEMPLATE) start_stage=0 ;;
+    TIMEPOINTS) start_stage=1 ;;
+    *) log_Err_Abort "Unrecognized option for start-stage: $StartStage"
+  esac
+fi
+end_stage=1
+if [ -n "$EndStage" ]; then  
+  case $EndStage in
+    TEMPLATE) end_stage=0 ;;
+    TIMEPOINTS) end_stage=1 ;;
+    *) log_Err_Abort "Unrecognized option for end-stage: $EndStage"
+  esac
+fi
+
+# ----------------------------------------------------------------------
+log_Msg "Starting main functionality"
+# ----------------------------------------------------------------------
+
+# ----------------------------------------------------------------------
+# Log values retrieved from command line options
+# ----------------------------------------------------------------------
+log_Msg "StudyFolder: ${StudyFolder}"
+log_Msg "SubjectID: ${SubjectID}"
+log_Msg "Sessions: ${Sessions}"
+log_Msg "TemplateID: ${TemplateID}"
+log_Msg "recon_all_seed: ${recon_all_seed}"
+log_Msg "StartStage: $StartStage"
+log_Msg "EndStage: $EndStage"
```

Comment:
This is still redundant, `opts_ShowValues` already logs these, please remove this block for brevity.